lib/game.rb

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env ruby
+require_relative '../lib/player.rb'
+class Game
+  def create_player(name, token, move_count)
+    player = Player.new(name, token, move_count)
+    player
+  end
+
+  # Display the game board
+  def display_board(board)
+    puts " #{board[0]}  | #{board[1]}  | #{board[2]}"
+    puts '-------------'
+    puts " #{board[3]}  | #{board[4]}  | #{board[5]}"
+    puts '-------------'
+    puts " #{board[6]}  | #{board[7]}  | #{board[8]}"
+  end
+
+  # player-1 selects a token and the other token is assigned to player-2
+  def select_tokens(player_one, player_two, game_tokens)
+    puts "#{player_one.name}, Choose a token: 'X' or 'O'"
+    selected_token = gets.chomp.upcase
+    loop do
+      if !game_tokens.include? selected_token
+        puts "Invalid token, please select 'X' or 'O'"
+        selected_token = gets.chomp.upcase
+      else
+        player_one.token = selected_token
+        player_two.token = selected_token == game_tokens[0] ? game_tokens[1] : game_tokens[0]
+
+        puts 'Good choice!!'
+        puts "#{player_one.name}'s Token:  #{player_one.token}"
+        puts "#{player_two.name}'s Token:  #{player_two.token}"
+        break
+      end
+    end
+  end
+
+  # returns board index where player's token is to be placed
+  def input_to_index(chosen_index, board)
+    # check token validity
+    valid = valid_move(chosen_index, board)
+
+    return chosen_index if valid
+  end
+
+  # checks if players made valid moves
+  def valid_move(index, _board)
+    if (1..9).include?(index)
+      true
+    else
+      false
+    end
+  end
+
+  # returns next token to be played
+  def switch_turn(player_one, player_two, last_token)
+    next_token = nil
+
+    if last_token.nil?
+      next_token = player_one.token
+    elsif last_token == player_one.token
+      next_token = player_two.token
+    elsif last_token == player_two.token
+      next_token = player_one.token
+    end
+
+    next_token
+  end
+
+  def check_board_full(board, game_tokens)
+    full = true
+
+    board.each do |item|
+      unless game_tokens.include? item
+        full = false
+        break
+      end
+    end
+
+    full
+  end
+
+  def check_winning(player, board, winning_combination)
```

Comment:
Please, remove `winning_combination` in parameters.
Move the `WINNING_COMBINATION` into that `Game` class as a constant